src/SFML/Window/Window.cpp

```diff
@@ -107,7 +107,7 @@ void Window::create(VideoMode mode, const String& title, Uint32 style, const Con
     // Check validity of style according to the underlying platform
     #if defined(SFML_SYSTEM_IOS) || defined(SFML_SYSTEM_ANDROID)
         if (style & Style::Fullscreen)
-            style &= ~Style::Titlebar;
+            style &= ~static_cast<Uint32>(Style::Titlebar);
```

Comment:
This is strange. The negation needs `static_cast`, but the logical or just below doesn't?

Reply:
Yeah, not sure why and I'm not even certain that this works correctly.
But you do have the different types of Uint32 for style and the enum being int, I assume by default.

Sounds to me like something we should align with C++17.

If you have a better way to do it let me know.

Reply:
With C++17 we would probably use `enum class`, and then you'd need `static_cast` all the time for flag-like enums.
One neat idiom I've seen is overloading `operator+`:

```cpp
// instead of
static_cast<sf::Uint32>(Style::Titlebar) | static_cast<sf::Uint32>(Style::Resize)

// you would have
+Style::Titlebar | +Style::Resize
```

Or directly overload the operator for the enum values. There are also libraries like [enum-flags](https://github.com/grisumbras/enum-flags) trying to address this problem. But for now I think it's OK to just silence the warnings.

---

It's a bit a shame that `static_cast` is such a horrible syntactical choice for something occurring relatively often. It really makes code unncessarily verbose.